code/modules/modular_computers/computers/modular_computer/variables.dm

```diff
@@ -0,0 +1,62 @@
+// This is the base type that handles everything. Subtypes can be easily created by tweaking variables in this file to your liking.
+
+/obj/item/modular_computer
+	name = "Modular Computer"
+	desc = "A modular computer. You shouldn't see this."
+
+	var/enabled = 0											// Whether the computer is turned on.
+	var/screen_on = 1										// Whether the computer is active/opened/it's screen is on.
+	var/datum/computer_file/program/active_program = null	// A currently active program running on the computer.
+	var/hardware_flag = 0									// A flag that describes this device type
+	var/last_power_usage = 0								// Last tick power usage of this computer
+	var/last_battery_percent = 0							// Used for deciding if battery percentage has chandged
+	var/last_world_time = "00:00"
+	var/list/last_header_icons
+	var/computer_emagged = FALSE							// Whether the computer is emagged.
+	var/apc_powered = FALSE									// Set automatically. Whether the computer used APC power last tick.
+	var/base_active_power_usage = 50						// Power usage when the computer is open (screen is active) and can be interacted with. Remember hardware can use power too.
+	var/base_idle_power_usage = 5							// Power usage when the computer is idle and screen is off (currently only applies to laptops)
+	var/bsod = FALSE										// Error screen displayed
+	var/ambience_last_played								// Last time sound was played
+
+	// Modular computers can run on various devices. Each DEVICE (Laptop, Console, Tablet,..)
+	// must have it's own DMI file. Icon states must be called exactly the same in all files, but may look differently
+	// If you create a program which is limited to Laptops and Consoles you don't have to add it's icon_state overlay for Tablets too, for example.
+
+	icon = null												// This thing isn't meant to be used on it's own. Subtypes should supply their own icon.
+	icon_state = null
+	center_of_mass = null									// No pixelshifting by placing on tables, etc.
+	randpixel = 0											// And no random pixelshifting on-creation either.
+	var/icon_state_unpowered = null							// Icon state when the computer is turned off
```

Comment:
That seems `icon_state_unpowered` is useless variable, what only increase icon_state related vars in two times.
Why not remove it (or update from actual `icon_state` in `Initialize()`)?

Reply:
removed

Reply:
removed it, it was obsolete